Sources/TestsCommon/CommonTestUtils.swift

```diff
@@ -136,12 +136,38 @@ open class MongoSwiftTestCase: XCTestCase {
 }
 
 /// Enumerates the different topology configurations that are used throughout the tests
-public enum TestTopologyConfiguration: String, Decodable {
+public enum TestTopologyConfiguration: RawRepresentable, Decodable {
     case sharded
-    case replicaSet = "replicaset"
+    case replicaSet
     case single
 
-    /// Determines the topologyType of a client based on the reply returned by running an isMaster command
+    public init?(rawValue: String) {
+        switch rawValue {
+        // as of MongoDB 3.6 (the lowest version this driver supports), shards must be replica sets, so for our
+        // testing purposes there is no distinction between these topology types.
```

Comment:
I thought this too when I was reviewing the Rust runner, but it seems like the docs might not be accurate here (https://github.com/mongodb/mongo-rust-driver/pull/249#discussion_r491147383).

Reply:
ah interesting. I originally changed the code to actually do a check by inspecting the contents of `config.shards`, but I deleted the code since it seemed superfluous. I guess I'll just revert my last commit then 🙂